docs/source/howto/codes.rst

```diff
@@ -4,40 +4,263 @@
+.. code-block:: python
+   :linenos:
+
+    from aiida.parsers.parser import Parser
+    from aiida import orm
+
+    class ArithmeticAddParser(Parser):
+
+        def parse(self, **kwargs):
+
+            output_folder = self.retrieved
+            output_filename = self.node.get_attribute('output_filename')
+
+            with output_folder.open(output_filename, 'r') as handle:
+                result = int(handle.read())
+
+            self.out('sum', orm.Int(result))
+
+The first command in the example (line 8) shows the way to get the ``retrieve`` folder that was generated by the associated |CalcJob| and that, by this point, should contain the files included in the ``retrieve_list`` (as specified in the |prepare_for_submission| method).
+The second one (line 9) is also accessing a parameter of the |CalcJob|: in this case, the name of the output.
+After unloading this information into local variables, these are then used to open said output file located in the obtained retrieved folder and read the single integer value that was written there by the addition |Code| (lines 11 and 12).
+
+Finally, it uses the ``out`` method to provide the ``sum`` output of the calculation its final value (of type AiiDA integer, as was specified in the |define| section of the associated |CalcJob|).
+It is important to note here that there is no return statement: the output is provided to the ``self.out`` method instead (any returned value is interpreted as an error signal) as an unstored node, and the engine will be in charge of performing the storing process.
+
+In order to use this ``ArithmeticAddParser`` inside an appropriate |CalcJob| (such as the one described in the :ref:`previous section <how-to:codes:interfacing>`), one needs to add it as a ``metadata.options.parser_name`` input.
+You can set a parser as the default option in the |define| method, but note that this choice can be overriden when instantiating the |CalcJob|.
+
+.. code-block:: python
+
+    @classmethod
+    def define(cls, spec):
+        (...)
+        spec.inputs['metadata']['options']['parser_name'].default = 'arithmetic.add'
+
+As can be seen in the previous line, the way to do this is not by passing the |Parser| class directly, but by providing the string label that identifies the |Parser| as a registered plugin in your working environment.
+In other words, in order to use a |Parser| you will need to register it as explained in the following how-to section on :ref:`registering plugins <how-to:plugins>`.
+
+Handling parsing errors
+-----------------------
+
+In order for you to be able to provide the user with information regarding the errors that can ocur after the calculation has finished (so, mostly during parsing), you have the option of defining ``exit_codes`` for cases when something goes wrong.
+The typical way to implement these is to use ``try``/``except`` clauses to wrap the lines of code that might raise some typical errors, that can then be replaced by these ``exit_codes``:
```

Comment:
I would not frame it as try/except because that is not all necessary or the idea behind. There don't need to be exceptions for one to want to return an exit code. For example you can check the output of the file, and all works just fine, but you conclude the output is not as desired, so you return an exit code. The try/except is not at all linked to in such a direct way. I would rather propose something like the following:
```suggestion
To signal a particular problem, you should just return the corresponding exit code.
These can be easily referenced in the parser through the ``self.exit_codes`` property.
For example, if the |CalcJob| defines an exit code with the label ``ERROR_READING_OUTPUT_FILE``, you can retrieve it as ``self.exit_codes.ERROR_READING_OUTPUT_FILE``.
If an exit code is returned from the parser, the engine will set it on the corresponding calculation job node.
```